librz/core/cmd/cmd_flirt.c

```diff
@@ -41,3 +41,73 @@ RZ_IPI RzCmdStatus rz_flirt_scan_handler(RzCore *core, int argc, const char **ar
 	rz_cons_printf("Found %d FLIRT signatures via %s\n", new - old, argv[1]);
 	return RZ_CMD_STATUS_OK;
 }
+
+RZ_IPI RzCmdStatus rz_flirt_function_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisFunction *func = rz_analysis_get_function_at(core->analysis, core->offset);
+	if (!func) {
+		RZ_LOG_ERROR("core: Cannot find function at 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	ut64 size = rz_analysis_function_size_from_entry(func);
+
+	ut8 *pattern = malloc(size);
+	if (!pattern) {
+		RZ_LOG_ERROR("core: cannot allocate function buffer.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (!rz_io_read_at(core->io, func->addr, pattern, (int)size)) {
+		RZ_LOG_ERROR("core: couldn't read function %s at 0x%" PFMT64x ".\n", func->name, func->addr);
+		free(pattern);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	ut8 *mask = rz_analysis_mask(core->analysis, size, pattern, func->addr);
+	if (!mask) {
+		RZ_LOG_ERROR("core: cannot allocate or calculate function mask.\n");
+		free(pattern);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	for (ut64 i = size - 1; i > 1; --i) {
+		if (mask[i] != 0xFF) {
+			size--;
+			continue;
+		}
+		break;
+	}
+
+	ut16 crc16 = 0;
+	ut32 crclen = 0;
+	if (size > RZ_FLIRT_MAX_PRELUDE_SIZE) {
+		// the crc should be generated only for when the buffer is > RZ_FLIRT_MAX_PRELUDE_SIZE
+		// also the size can be zero if after the prelude there is only masked bytes
+		crclen = rz_sign_flirt_crc16_length(mask + RZ_FLIRT_MAX_PRELUDE_SIZE, size - RZ_FLIRT_MAX_PRELUDE_SIZE);
+		crc16 = rz_sign_flirt_crc16(pattern + RZ_FLIRT_MAX_PRELUDE_SIZE, crclen);
+	}
```

Comment:
having this here seems error prone. I'd expect this kind of low-level details about how flirt works to be hidden from "users" of the flirt API.